lib/core/importer.ts

```diff
@@ -44,21 +43,22 @@ export class Importer {
 
   private scan(pathName: string): AutoClasses {
     const exports: Record<string, unknown> = require(pathName);
-    const autoClasses = Object.values(exports).filter((value) => typeof value === 'function') as ClassType[];
+    return (Object.values(exports) as ClassType[]).reduce(
+      (classes: AutoClasses, value: ClassType) => {
+        if (typeof value === 'function') {
+          Reflect.hasMetadata(COMPONENT_SCAN_WATERMARK, value) && this.catchOverlappedScanScope(value, pathName);
```

Comment:
Why you use just boolean return function `hasMetadata` with no references? (e.g. if  or  catching values )

Reply:
I want to call `this.catchOverlappedScanScope(value, pathName)` when `Reflect.hasMetadata(COMPONENT_SCAN_WATERMARK, value)` value is true. Is there anything wrong with it? I don't get it. 🥲

Reply:
it seems to more readable by using `if block` 🤣

Reply:
`Reflect.hasMetadata(COMPONENT_SCAN_WATERMARK, value) && this.catchOverlappedScanScope(value, pathName);`
What I want to represent through code is above statement has different purpose from below ìf `blocks`.

Reply:
still bad with the commented reason? 🥹

Reply:
I couldn't think about your philosophical think by just reading that code 😭 
After reading your additional comment, i just understand it.
In long range-sight for code lifecycle for project, if it's possible avoiding these unusual code style is better.

overall, approved it for now.

Reply:
Thank you !